src/utils.c

```diff
@@ -7,11 +8,50 @@ bool is_bool(SEXP x) {
     *LOGICAL(x) != NA_LOGICAL;
 }
 
-SEXP vctrs_dispatch3(SEXP fn, SEXP x, SEXP y) {
-  SEXP dispatch_call = PROTECT(Rf_lang3(fn, x, y));
-  SEXP out = Rf_eval(dispatch_call, R_GlobalEnv);
+/**
+ * Dispatch with two arguments
+ *
+ * @param fn The method to call.
+ * @param x,y Arguments passed to the method.
+ * @param fn_sym,x_sym,y_sym Symbols to which `x` and `y` should be
+ *   assigned.  The assignment occurs in `env` and the dispatch call
+ *   refers to these symbols.
+ * @param env The environment in which to dispatch. Should be the
+ *   global environment or inherit from it so methods defined there
+ *   are picked up. If the global environment, a child is created so
+ *   the call components can be masked.
+ *
+ *   If `env` contains dots, the dispatch call forwards dots.
+ */
+SEXP vctrs_dispatch2(SEXP fn_sym, SEXP fn,
```

Comment:
This is to generate a better traceback?

Reply:
And to forward dots if the environment has dots.